src/lib/HST/Frontend/Syntax.hs

```diff
@@ -438,3 +442,27 @@ instance Eq (SrcSpan a) where
 -- | Custom order for 'SrcSpan' which treats all source spans as equal.
 instance Ord (SrcSpan a) where
   _ `compare` _ = EQ
+
+-- | Type for storing basic data of a source span, i. e. the file path and the
+--   line and column of the start and end of the spanned source code.
+--
+--   The data contained in this type is used for displaying input code
+--   excerpts. This additional type is necessary in order to have a type for
+--   source spans without type variables, which would cause problems in the
+--   "HST.Effect.Report" module.
+data MsgSrcSpan = MsgSrcSpan
+  { msgSrcSpanFilePath    :: FilePath
+  , msgSrcSpanStartLine   :: Int
+  , msgSrcSpanStartColumn :: Int
+  , msgSrcSpanEndLine     :: Int
+  , msgSrcSpanEndColumn   :: Int
+  }
+ deriving ( Eq, Show )
```

Comment:
I think the only reason why the `MsgSrcSpan` data type cannot be in the same module as the remaining message code is that the `message` smart constructor takes a `S.SrcSpan`. By adding a type class for `toMsgSrcSpan`, this dependency could be eliminated.

Reply:
I would leave the `MsgSrcSpan` in this location, because I like that the `Syntax` module currently has no imports and stores the entirety of our ASTs in types defined in the module itself, except for the types represented by the type families. And I also think that `MsgSrcSpan` fits into the `Syntax` module: It was only created because there were problems with having a type variable in the `SrcSpan` type and I don't think this solution to these problems should cause the module this data is stored in to be changed.
However, if you really want to have `MsgSrcSpan` in the new `Message` module, I will move it.